### PR TITLE
Addl fields demo fixes

### PR DESCRIPTION
**Pull request by gcsantos-gpa**

### CODE REVIEW

**Review by clackner-gpa — CHANGES_REQUESTED** [review]
Since `_` and spaces are now treated the same we may want to adjust Line 120 of `AdditionalFieldForm.tsx` to include `replace('_',' ')` or something along those lines to avoid confusion

Or a;lternatively just add an error case if the Field includes `_` at all....

**Comment by gcsantos-gpa** [comment]
> Since `_` and spaces are now treated the same we may want to adjust Line 120 of `AdditionalFieldForm.tsx` to include `replace('_',' ')` or something along those lines to avoid confusion
> 
> Or a;lternatively just add an error case if the Field includes `_` at all....

Maybe the solution it to add this rule to the check on name uniqueness? I think perserving the ability to make fields either "field_name" or "field name" is important due to how old data could be either.

**Comment by clackner-gpa** [comment]
Sounds like adding `.replace('_',' ')` on both sides of the `=` should do it then. that way `field_name` and `field name` are considered equal and will throw the error in the UI

File: Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/AdditionalFields/AdditionalFieldForm.tsx [hunk 1]
```diff
@@ -115,22 +115,23 @@ export default function AdditionalFieldForm(props: IProps) {
...
         props.SetErrors(e);
-    }, [props.Record]);
+    }, [props.Record, extDBId]);
 
     React.useEffect(() => {
         if (props.SetWarnings === undefined) return;
         let w = [];
-        if (props.Record.IsKey && allAddlFields.findIndex((d) => d.ID != props.Record.ID && d.IsKey) !== -1)
+        if (props.Record.IsKey && allAddlFields.findIndex((d) => (d.ID != props.Record.ID) && (d.ExternalDBTableID == props.Record.ExternalDBTableID) && d.IsKey) > -1)
             w.push(`A key field already exists for this External Database.`);
```
Comment (gcsantos-gpa) [inline Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/AdditionalFields/AdditionalFieldForm.tsx:134]:
```suggestion
            w.push(`A key field already exists for this External Table.`);
```

**Review by clackner-gpa — APPROVED** [review]

**Pull request merged**